### PR TITLE
Include inherited members in Sphinx Docs

### PR DESCRIPTION
**Pull request by awonak**
Staged changes here: https://awonak.github.io/EuroPi/api.html

This moves the API docs onto the EuroPi/api.html page instead of EuroPi/generated/europi.html.

Also added a few docstring fixes to correct documentation.

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -1,7 +1,5 @@
 API
 ===
 
-.. autosummary::
-   :toctree: generated
-
-   europi
\ No newline at end of file
+.. automodule:: europi
+   :inherited-members:
```
Comment (mjaskula) [inline docs/api.rst:5]: I think that you want to leave this file as is, and add the `:inherited-members:` config to `_templates/autosummary/module.rst`. Both of these changes will make it easier to add additional files in the future.

**Review by mjaskula — CHANGES_REQUESTED** [review]

File: docs/api.rst (hunk 1 above)
Comment (awonak) [reply docs/api.rst]: Thanks for the tip, that worked great!

https://awonak.github.io/EuroPi/generated/europi.html#module-europi

File: .github/workflows/publish_docs.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ on:
   push:
     branches:
       - main
+      - docs-inherited
```
Comment (mjaskula) [inline .github/workflows/publish_docs.yml:9]: Can be removed before merge.

**Comment by mjaskula** [comment]
Do you intend to remove the copied text from `software/README.md` in this PR? I think you should. Just add a link to the published API docs.

**Review by mjaskula — APPROVED** [review]

**Pull request merged**